src/yang/ietf-bgp-common.yang

```diff
@@ -322,6 +327,15 @@ submodule ietf-bgp-common {
         "RFC 2385: Protection of BGP Sessions via the TCP MD5
                    Signature Option.";
     }
+    container bfd {
```

Comment:
Are you proposing to make the bfd container part of the neighbor-group-transport-config grouping? If so, and just to let you know, that grouping is used not only in the main BGP module neighbor config, but also in the peer group configuration. If that is not intended, why not keep the container under the neighbor config in the main BGP module, like it was before.

Reply:
BFD configuration should be part of group configuration.  As part of trying to understand if that was an issue, I opened issue 154 that notes other discrepancies like the security settings.

While it's possible for security mechanisms to need to be per-peer, in many cases they are per-group and selectively overridden if it is necessary.

Reply:
So you are suggesting that BFD configuration applies to peer group configuration also. How would that work? I mean, if I have configured BFD with some parameters at the local end, how am I conveying them to the peer? Maybe I am missing something here.

Reply:
> So you are suggesting that BFD configuration applies to peer group configuration also. How would that work? I mean, if I have configured BFD with some parameters at the local end, how am I conveying them to the peer? Maybe I am missing something here.

Remember, this is the client's desired usage for BFD.  A peer-group is just a template that is shared by a number of neighbors.  Thus, saying "yes, I want bfd enabled, local-multiplier 3, 10ms interval" for the group is appropriate.

A peer group doesn't instantiate peering.  A neighbor must be configured that uses the peer-group. See /bgp/neighbors[ip]/peer-group leaf.